crates/core/c8y_api/src/json_c8y.rs

```diff
@@ -286,4 +344,109 @@ mod tests {
             EMPTY_STRING
         );
     }
+
+    #[test_case(
+        ThinEdgeEvent {
+            name: "click_event".into(),
+            data: Some(ThinEdgeEventData {
+                text: Some("Someone clicked".into()),
+                time: Some(datetime!(2021-04-23 19:00:00 +05:00)),
+                extras: HashMap::new(),
+            }),
+        },
+        C8yCreateEvent {
+            source: None,
+            event_type: "click_event".into(),
+            time: datetime!(2021-04-23 19:00:00 +05:00),
+            text: "Someone clicked".into(),
+            extras: HashMap::new(),
+        }
+        ;"event translation"
+    )]
+    #[test_case(
+        ThinEdgeEvent {
+            name: "click_event".into(),
+            data: Some(ThinEdgeEventData {
+                text: None,
+                time: Some(datetime!(2021-04-23 19:00:00 +05:00)),
+                extras: HashMap::new(),
+            }),
+        },
+        C8yCreateEvent {
+            source: None,
+            event_type: "click_event".into(),
+            time: datetime!(2021-04-23 19:00:00 +05:00),
+            text: "click_event".into(),
+            extras: HashMap::new(),
+        }
+        ;"event translation without text"
+    )]
+    #[test_case(
+        ThinEdgeEvent {
+            name: "click_event".into(),
+            data: Some(ThinEdgeEventData {
+                text: Some("Someone, clicked, it".into()),
+                time: Some(datetime!(2021-04-23 19:00:00 +05:00)),
+                extras: HashMap::new(),
+            }),
+        },
+        C8yCreateEvent {
+            source: None,
+            event_type: "click_event".into(),
+            time: datetime!(2021-04-23 19:00:00 +05:00),
+            text: "Someone, clicked, it".into(),
+            extras: HashMap::new(),
+        }
+        ;"event translation with commas in text"
+    )]
+    fn check_event_translation(
+        tedge_event: ThinEdgeEvent,
+        expected_c8y_event: C8yCreateEvent,
+    ) -> Result<()> {
+        let actual_c8y_event = C8yCreateEvent::try_from(tedge_event)?;
+
+        assert_eq!(expected_c8y_event, actual_c8y_event);
+
+        Ok(())
+    }
+
+    #[test]
+    fn event_translation_empty_json_payload_generates_timestamp() -> Result<()> {
+        let tedge_event = ThinEdgeEvent {
+            name: "empty_event".into(),
+            data: Some(ThinEdgeEventData {
+                text: None,
+                time: None,
+                extras: HashMap::new(),
+            }),
+        };
+
+        let actual_c8y_event = C8yCreateEvent::try_from(tedge_event)?;
+
+        assert_eq!(actual_c8y_event.event_type, "empty_event".to_string());
+        assert_eq!(actual_c8y_event.text, "empty_event".to_string());
+        assert_matches!(actual_c8y_event.time, _);
```

Comment:
Ha ha that was silly 😜 . The `time` field used was an `Option<OffsetDateTime>` earlier and then my assertion was matching for `Some(_)` time value. But, `time` is not an `Option` anymore but just an `OffsetDateTime`. So I guess this assertion doesn't mean anything and hence I'll remove it. I can't assert the precise value of that `OffsetDateTime` because it's generated internally by the mapper. I just has to be something.

Reply:
> So I guess this assertion doesn't mean anything 

This simply asserts that `time` matches `_`, something that is always true. So meaningful but useless.

Reply:
This assert can be removed, since bringing no value.